changelog/2024-04-08T06_51_45+00_00_smap_with_bounds

```diff
@@ -0,0 +1 @@
+* Add proof witness to `dfold` and `smapWithBounds`
```

Comment:
```suggestion
* CHANGED: Add proof witness to `dfold` and `smapWithBounds`
```

For earlier (Slack?) discussion I understood that this can sometimes cause code not to compile due to ambiguous variables. Is this true? If so, this heads up should be added to the change log.